libcontainer/configs/validate/rootless.go

```diff
@@ -104,11 +95,24 @@ func rootlessMount(config *configs.Config) error {
 		// Check that the options list doesn't contain any uid= or gid= entries
 		// that don't resolve to root.
 		for _, opt := range strings.Split(mount.Data, ",") {
-			if strings.HasPrefix(opt, "uid=") && opt != "uid=0" {
-				return fmt.Errorf("cannot specify uid= mount options in rootless containers where argument isn't 0")
+			if strings.HasPrefix(opt, "uid=") {
+				uid, _ := strconv.Atoi(strings.Replace(opt, "uid=", "", 1))
```

Comment:
I don't like the ignoring of errors, nor `strings.Replace` here. Can we just do something like

```
var uid int
n, err := fmt.Sscanf(opt, "uid=%d", &uid)
if n != 1 || err != nil {
    // Ignore unknown mount options.
    continue
}
if !hasIDMapping(uid, config.UidMappings) {
    return fmt.Errorf("cannot specify uid= mount options for unmapped uid in rootless containers")
}
```

And the same for the other option.